src/node.rs

```diff
@@ -1229,6 +1229,24 @@ impl<S: ForkSource + std::fmt::Debug> InMemoryNode<S> {
             inner.filters.notify_new_pending_transaction(tx_hash);
         }
 
+        let execution_mode = if self
+            .inner
+            .read()
+            .map_err(|e| format!("Failed to acquire read lock: {}", e))?
+            .impersonated_accounts
+            .contains(&l2_tx.common_data.initiator_address)
+        {
+            tracing::info!(
+                "🕵️ Executing tx from impersonated account {:?}",
+                l2_tx.common_data.initiator_address
+            );
+            // Using the eth call here to avoid all the account checks
+            // TODO: think about fictive blocks in case of impersonating via eth call
+            TxExecutionMode::EthCall
```

Comment:
the alternative, is to use the other DefaultAccount implementation in such case.

(as EthCall is in theory not supposed to be used for blocks that we commit)

Reply:
ah right, you actually want to have even more permissions...

The cool thing, is that we have a local copy of bootloader, so we can add some additional modes to do exactly that - and we can be explicit about it (instead of depending on a lucky side effect of EthCall..)

Reply:
Agreed, I was planning to do that with preprocessing, but for now that's just a quick solution